__tests__/app.test.js

```diff
@@ -195,3 +195,82 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+describe("POST /api/articles/:article_id/comments", () => {
+  test("responds with the posted comment", () => {
+    const comment = { username: "butter_bridge", body: "Hello world!" };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(comment)
+      .expect(201)
+      .then(({ body }) => {
+        const postedComment = body.comment;
+
+        expect(typeof postedComment.comment_id).toBe("number");
+        expect(typeof postedComment.body).toBe("string");
+        expect(typeof postedComment.article_id).toBe("number");
+        expect(typeof postedComment.author).toBe("string");
+        expect(typeof postedComment.votes).toBe("number");
+        expect(typeof postedComment.created_at).toBe("string");
+      });
+  });
+  test("responds with status 400 and 'bad request' if username does not exist in database", () => {
+    const comment = { username: "banana", body: "Hello world!" };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(comment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("bad request");
+      });
+  });
+  test("responds with status 400 and 'bad request' if comment body is undefined", () => {
+    const comment = { username: "butter_bridge", body: undefined };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(comment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("bad request");
+      });
+  });
+  test("responds with status 400 and 'bad request' if article_id endpoint is valid but non-existent", () => {
+    const comment = { username: "butter_bridge", body: "Hello world!" };
+    return request(app)
+      .post("/api/articles/9999/comments")
+      .send(comment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("bad request");
+      });
+  });
+  test("responds with status 400 and 'bad request' if article_id endpoint is invalid", () => {
+    const comment = { username: "butter_bridge", body: "Hello world!" };
+    return request(app)
+      .post("/api/articles/banana/comments")
+      .send(comment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("bad request");
+      });
+  });
+  test("responds with status 404 and 'endpoint not found!' if comment endpoint is invalid", () => {
+    const comment = { username: "butter_bridge", body: "Hello world!" };
```

Comment:
This is a bad path test, not related to this particular endpoint so doesn't belong in this describe block for this endpoint. You may have already tested a bad path and so don't need to do it again. If our test auto passed that's usually a good indicator for us to rethink our test.